apps/fxc-front/src/app/logic/score/scorer.ts

```diff
@@ -1,19 +1,104 @@
-import { CircuitType } from '@flyxc/optimizer/lib/api';
-
-export class Score {
-  distanceM: number;
-  indexes: number[];
-  multiplier: number;
-  circuit: CircuitType;
-  closingRadiusKm: number;
-  points: number;
-
-  constructor(score: Partial<Score>) {
-    this.distanceM = score.distanceM ?? 0;
-    this.indexes = score.indexes ?? [];
-    this.multiplier = score.multiplier ?? 1;
-    this.circuit = score.circuit ?? CircuitType.OpenDistance;
-    this.closingRadiusKm = score.closingRadiusKm ?? 0;
-    this.points = score.points ?? 0;
+import type { LatLonAltTime, ScoringResult } from '@flyxc/optimizer/lib/optimizer';
+
+import { type Request as WorkerRequest, type Response as WorkerResponse } from '../../workers/optimizer';
+import ScoringWorker from '../../workers/optimizer?worker';
+import { getScoringRuleName, type LeagueCode } from './league/leagues';
+
+export enum ScoreOrigin {
+  INTERACTIVE = 'interactive',
```

Comment:
I think code in this folder should not deal with UI concerns but should rather be independent.
I don't think interactive belongs to here.
But le me revisit once I have time to review thoroughly

Reply:
I agree. I suppose these concepts could be in a better place in state management (planner-slice)

Reply:
Here, Score could be renamed: may be ScoringInfo? or ScoringContext? other?

Reply:
I think we should drop that origin and add a mechanism local to the components if needed but hopefully we won't need it.

Reply:
moved Score and ScoreOrigin to redux/planner-slice.ts